python/cleanup_CNRM_HistC_atlas_netcdf.ipynb

```diff
@@ -0,0 +1,1614 @@
+       "           2.54692118e-01, 9.51821709e-01, 6.63924768e-01],\n",
+       "          [9.16772040e-01, 7.16553571e-01, 8.98055176e-01, ...,\n",
+       "           4.49716512e-01, 7.01903567e-02, 5.32627750e-01],\n",
+       "          ...,\n",
+       "          [2.52715178e-01, 6.61247501e-02, 2.89715573e-01, ...,\n",
+       "           9.35940004e-01, 8.62441119e-01, 3.31952051e-01],\n",
+       "          [3.94367279e-01, 1.07535501e-01, 4.40893499e-01, ...,\n",
+       "           9.96838457e-01, 4.82223694e-01, 4.52619565e-01],\n",
+       "          [1.96022273e-01, 1.17496735e-01, 4.63722811e-01, ...,\n",
+       "           7.02232810e-01, 7.60008920e-01, 5.30220450e-01]]]]])</pre></div></li></ul></div></li><li class='xr-section-item'><input id='section-4fa0445d-a33e-4591-a7a8-d9ef32b7a938' class='xr-section-summary-in' type='checkbox'  checked><label for='section-4fa0445d-a33e-4591-a7a8-d9ef32b7a938' class='xr-section-summary' >Attributes: <span>(1)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><dl class='xr-attrs'><dt><span>description :</span></dt><dd>CNRM HistC data.</dd></dl></div></li></ul></div></div>"
+      ],
+      "text/plain": [
+       "<xarray.Dataset>\n",
+       "Dimensions:      (season: 2, constrained: 2, percentile: 4, lat: 19, lon: 20)\n",
+       "Coordinates:\n",
+       "  * season       (season) <U3 'DJF' 'JJA'\n",
+       "  * constrained  (constrained) int64 1 0\n",
+       "  * percentile   (percentile) int64 10 25 75 90\n",
+       "  * lat          (lat) float64 31.25 33.75 36.25 38.75 ... 71.25 73.75 76.25\n",
+       "  * lon          (lon) float64 -8.75 -6.25 -3.75 -1.25 ... 33.75 36.25 38.75\n",
+       "Data variables:\n",
+       "    tas          (season, constrained, percentile, lat, lon) float64 0.5703 ....\n",
+       "Attributes:\n",
+       "    description:  CNRM HistC data."
+      ]
+     },
+     "metadata": {},
+     "execution_count": 6
+    }
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "source": [
+    "# assembly data\n",
+    "def assembly(ds_original, ds_target, var, season, constrained, percentile):\n",
+    "    \"\"\"\n",
+    "    Select data from original nc files and save the target fields\n",
+    "    \n",
+    "    \"\"\"\n",
+    "    cons = [\"uncons\",\"cons\"] # 0: unconstrained 1: constrained\n",
+    "    seasons = [\"DJF\", \"JJA\"]\n",
+    "    key_p = dict(zip(percentile, range(len(percentile))))\n",
+    "    key_c = dict(zip(constrained, range(len(constrained))))\n",
+    "    key_s = dict(zip(seasons, range(len(seasons))))\n",
+    "    for c in constrained:\n",
+    "        for p in percentile:\n",
+    "            # due to the lon from 0-360 to -180-180\n",
+    "            ds_target[f\"{var}\"].values[key_s[season],key_c[c],key_p[p],:,:4] = ds_original[f\"q{p}_{cons[c]}\"].values[48:67, -4:]\n",
```

Comment:
This manual indexing is a bit itchy, especially the `[48:67, -4:]`. It would be more readable and less error-prone if we could for example use:

- `xr.concat([ds_10perc, ds_20perc, ds_30perc], concat_dim=percentile)` etc
- `ds.sel(lon=Slice(-10, 40), lat=Slice(30, 60), method='nearest')  # select Europe only`
- ` ds['lon'] = (ds['lon'] + 180) % 360 - 180; ds = ds.sortby(ds.lon)  # https://stackoverflow.com/a/53561230`

Reply:
@Peter9192 Thanks for your suggestion. Indeed I'm not so familiar with `xarray` and therefore was not aware of those clean data handling operations within xarray. I refactor the code using the 'selecting' approach within `xarray` and now the indexing is not manually selected.